cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -267,3 +269,14 @@ func ValidateMixedArguments(flag *pflag.FlagSet) error {
 	}
 	return nil
 }
+
+func ValidateAPIEndpoint(c *kubeadm.API, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	endpoint := util.GenerateMasterEndpoint(c.AdvertiseAddress, c.BindPort)
```

Comment:
Right, but given the code proposed, that can only fail, if the IP address is not valid. So, instead of creating an endpoint and then reporting on whether or not it was successful, one could check that IP is valid, and if so, create the endpoint using JoinHostPort(). IOW validating the IP infers the URL will be valid.